src/modules/compass/MeasurementTemplatesModal.js

```diff
@@ -0,0 +1,312 @@
+          text: 'OK',
+          onPress: () => deleteTemplate(selectedTemplate),
+        },
+      ],
+      {cancelable: false},
+    );
+  };
+
+  const createNewOrEditTemplate = (mode) => {
+    setDisplayForm(true);
+    if (mode === 'new') {
+      setName('');
+      setSelectedTemplate({values: {}});
+    }
+    if (tabularOrientation) setFormName(['measurement_bulk', 'tabular_orientation']);
+    else setFormName(['measurement_bulk', props.type]);
+    props.type === 'planar_orientation' ? dispatch(
+      setModalVisible({modal: MODALS.NOTEBOOK_MODALS.MEASUREMENT_PLANAR_TEMPLATE_FORM})) : dispatch(
+      setModalVisible({modal: MODALS.NOTEBOOK_MODALS.MEASUREMENT_LINEAR_TEMPLATE_FORM}));
+  };
+
+  const deleteTemplate = (template) => {
+    const existingTemplatesButCurrentTemplate = measurementTemplates.filter(
+      templateId => selectedTemplate.id != templateId.id);
+    dispatch(addedMeasurementTemplates(existingTemplatesButCurrentTemplate));
+    if ((!isEmpty(activeMeasurementTemplates[0]) && activeMeasurementTemplates[0].id === template.id) || (!isEmpty(
+      activeMeasurementTemplates[1]) && activeMeasurementTemplates[1].id === template.id)) {
+      clearTemplate();
+    }
+    closeMeasurementTemplatesModal();
+  };
+
+  const editTemplate = (template) => {
+    setSelectedTemplate(template);
+    setName(template.name);
+    createNewOrEditTemplate('edit');
+  };
+
+  function getRelevantMeasurementTemplates(type) {
+    return !isEmpty(measurementTemplates) ? measurementTemplates.filter(
+      template => template.type === type) : [];
+  }
+
+  const onSwitchPlanarTabular = (i) => {
+    const currentType = tabularOrientation ? 'tabular_orientation' : 'planar_orientation';
+    if ((i === 0 && currentType === 'tabular_orientation') || (i === 1 && currentType === 'planar_orientation')) {
+      const newType = currentType === 'tabular_orientation' ? 'planar_orientation' : 'tabular_orientation';
+
+      const typeText = newType === 'tabular_orientation' ? 'Tabular Zone' : 'Planar Orientation';
+      if (!isEmpty(selectedTemplate.id)) {
+        const alertTextEnd = 'this measurement to a ' + typeText + '? You will '
+          + 'lose all data for the current template.';
+        Alert.alert('Switch to ' + typeText, 'Are you sure you want to switch ' + alertTextEnd,
+          [{
+            text: 'Cancel',
+            onPress: () => console.log('Cancel Pressed'),
+            style: 'cancel',
+          }, {
+            text: 'OK',
+            onPress: () => newType === 'tabular_orientation' ? setTabularOrientation(true) : setTabularOrientation(
+              false),
+          }],
+          {cancelable: false},
+        );
+      }
+      else newType === 'tabular_orientation' ? setTabularOrientation(true) : setTabularOrientation(false);
+    }
+  };
+
+  const renderFormFields = () => {
+    return (
+      <View style={{flex: 1}}>
+        {props.type === 'planar_orientation'
+        && <ButtonGroup
```

Comment:
use `(` then drop `<ButtonGroup` to next line so it can line up with closing element